src/utils.ts

```diff
@@ -132,3 +132,16 @@ export const isDefined = (value: any): boolean => value !== void 0;
 
 export const isMatchingPath = (path: string, criteria: string | RegExp) =>
     isRegex(criteria) ? !!path.match(criteria as RegExp) : checkGlob(path, criteria as string);
+
+export const isExpanded = (path: string, state: JsonViewerState): boolean => state.expanded[path];
+
+export const isCollapsed = (path: string, state: JsonViewerState): boolean => !state.expanded[path];
+
+export const isDeepestVisible = (node: HTMLElement): boolean => node.children[1] instanceof HTMLSpanElement;
```

Comment:
same as above about querying into specific places like `children[1]`; feels especially suspicious inside of a utils module. i guess i'm suggesting moving `isDeepestVisible` to the keyboard module, and `isExpandedAndNonPrimitive` to the JsonViewer?

Reply:
Do you feel better about these now that I'm not accessing the array's directly?

Reply:
yiss